### ISSUE TITLE
Support layered output for KML

### ISSUE THREAD

**Issue by tmcw** [placemark/togeojson#57]
KML supports layers. toGeoJSON ignores then and just gives you all the features in your KML file. 

Strawman draft:

```js
output = toGeoJSON.kmlLayered(xmlDocument);
output = [
{
  layer: 'foo',
  geojson: { type: 'FeatureCollection', /* … */ }
},
{
  layer: 'bar',
  geojson: { type: 'FeatureCollection', /* … */ }
}
]
```

**Comment by throberto** [comment]
layer name could not be added as collection feature properties?

example:
```
{
  "type": "FeatureCollection",
  "features": [
    {
      "type": "Feature",
      "properties": {
         "kmlLayerName": "foo"
       },
      "geometry": {
        "type": "Point",
        "coordinates": [
          83.3203125,
          -27.371767300523032
        ]
      }
    }
  ]
}
```

**Comment by tmcw** [comment]
Unfortunately I don't think that would be sufficient - layers have metadata of their own - name, description, etc. They can also be nested, and the order of features in and amongst folders might be important.

5.x, which is now in master, implements a nested tree structure that adds folder support.

**Comment by tmcw** [comment]
This is released as a major version in 5.0.0, with a nested folder structure!
